### ISSUE TITLE
The new toMatchSnapshot interferes with jest-snapshot 

### ISSUE THREAD

**Issue by paymand** [webdriverio/expect-webdriverio#1453]
Works: expect-webdriverio@4.2.7
Broken: expect-webdriverio@4.11.2
Most probable culprit: https://github.com/webdriverio/expect-webdriverio/pull/1426

Background:

We have a custom matcher `toBeAccessible` to check for a11y using axe-core which saves the violations as jest snapshots:

`wdio.conf.ts`:
```js
  before: async function () {
     setOptions({ wait: 20000 });
     await extendWdioCommands(browser);
     // Workaround for https://github.com/webdriverio/expect-webdriverio/pull/797
     // @ts-expect-error fixme
     if (global.expect.expect !== undefined) {
       // @ts-expect-error fixme
       global.expect = global.expect.expect;
     }

     matchers.addCustomMatchers();
   },
```
`matchers.js`:
```js
import path from 'path';
import axeCore from 'axe-core';
import { SnapshotState, toMatchSnapshot } from 'jest-snapshot';

function formatViolations(violations) {
  return violations.map((violation) => {
    return {
      id: violation.id,
      help: violation.help,
      nodes: violation.nodes.map((node) => node.target.toString()),
    };
  });
}

async function getAxeContext(object) {
  if (['Element', 'String'].includes(object.constructor.name)) {
    return object;
  }
  if (object.rootElement) {
    return await object.rootElement;
  }
  return undefined;
}

export function addCustomMatchers() {
  expect.extend({
    async toBeAccessible(element, options = {}) {
      const {
        snapshotName = `[${browser.capabilities.browserName}-${browser.requestedCapabilities.browserVersion}-${browser.capabilities.browserVersion}] ${test.parent} ${test.title}`,
        update = process.env.JEST_SNAPSHOT_UPDATE_ALL,
        ci = process.env.CI,
      } = options;

      const testFileDir = path.dirname(test.file);
      const testFileBaseName = path.basename(test.file);
      const snapshotPath = path.join(
        testFileDir,
        '__snapshots__',
        `${testFileBaseName}.snap`,
      );

      const context = await getAxeContext(await element);
      // add the axe script
      await browser.execute(axeCore.source);
      await browser.pause(1000); // FIXME: find a better way to wait for all animations / transitions to finish
      const results = await browser.executeAsync(
        function (browserContext, axeRunOptions, done) {
          const ctx = browserContext || document;
          window.axe.run(
            ctx,
            Object.assign(
              {},
              {
                runOnly: {
                  type: 'tag',
                  values: ['wcag2a'],
                },
              },
              axeRunOptions,
            ),
            (err, actionResults) => {
              if (actionResults) {
                actionResults.violations.forEach((violation) => {
                  const selectors = violation.nodes
                    .map((node) => node.target.toString())
                    .join(',');
                  document.querySelectorAll(selectors).forEach(function (e) {
                    e.style['background-color'] = 'magenta';
                  });
                });
              }
              done(err || actionResults);
            },
          );
        },
        context,
        options,
      );

      const violations = formatViolations(results.violations);

      const snapshotState = new SnapshotState(snapshotPath, {
        updateSnapshot: ci ? 'none' : update ? 'all' : 'new',
        prettierPath: undefined,
        snapshotFormat: {},
      });

      const result = toMatchSnapshot.call(
        Object.assign({}, this, {
          currentTestName: snapshotName,
          snapshotState,
        }),
        violations,
      );
      snapshotState.save();
      if (
        result.error &&
        result.error.message &&
        /expected value to match snapshot/.test(result.error.message)
      ) {
        result.error.stack = `\nExpected:\n${result.error.expected}\n--\nActual:\n${result.error.actual}\n--\n${result.error.stack}`;
      }
      return result;
    },
  });
}
```
we have been using this matcher for ages and we have jest snapshots created such as this one:
```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[chrome-desktop-119.0.6045.199] <test name>`] = `
Array [
  Object {
    "help": "Required ARIA attributes must be provided",
    "id": "aria-required-attr",
    "nodes": Array [
    ...
```

Ever since the update to expect-webdriverio@4.11.2 (mind you, it's a minor update 😢) the newly added snapshot functionality messes up with our jest snapshot:

```
@@ -1,4 +1,4 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
+// Snapshot v1
```

**Comment by paymand** [comment]
I ended up replacing `jest-snapshot` with WebdriverIO's [toMatchSnapshot](https://github.com/webdriverio/expect-webdriverio/blob/main/docs/API.md#snapshot-matcher):

```js
import axeCore from 'axe-core';

function formatViolations(violations) {
  return violations.map((violation) => {
    return {
      id: violation.id,
      help: violation.help,
      nodes: violation.nodes.map((node) => node.target.toString()),
    };
  });
}

async function getAxeContext(object) {
  if (['Element', 'String'].includes(object.constructor.name)) {
    return object;
  }
  if (object.rootElement) {
    return await object.rootElement;
  }
  return undefined;
}

async function getAccessiblityViolations(element, options = {}) {
  const context = await getAxeContext(await element);
  // add the axe script
  await browser.execute(axeCore.source);
  await browser.pause(1000); // FIXME: find a better way to wait for all animations / transitions to finish
  const results = await browser.executeAsync(
    function (browserContext, axeRunOptions, done) {
      const ctx = browserContext || document;
      window.axe.run(
        ctx,
        Object.assign(
          {},
          {
            runOnly: {
              type: 'tag',
              values: ['wcag2a'],
            },
          },
          axeRunOptions,
        ),
        (err, actionResults) => {
          if (actionResults) {
            actionResults.violations.forEach((violation) => {
              const selectors = violation.nodes
                .map((node) => node.target.toString())
                .join(',');
              document.querySelectorAll(selectors).forEach(function (e) {
                e.style['background-color'] = 'magenta';
              });
            });
          }
          done(err || actionResults);
        },
      );
    },
    context,
    options,
  );

  return formatViolations(results.violations);
}

export function addCustomMatchers() {
  expect.extend({
    async toBeAccessible(element, options = {}) {
      const violations = await getAccessiblityViolations(element, options);

      const snapshotName = `[${browser.capabilities.browserName}-${
        global.isMobile ? 'mobile' : 'desktop'
      }]`;
      await expect(violations).toMatchSnapshot(snapshotName);

      return { pass: true };
    },
  });
}
```

**Comment by christian-bromann** [comment]
> Ever since the update to expect-webdriverio@4.11.2 (mind you, it's a minor update 😢) the newly added snapshot functionality messes up with our jest snapshot:

We have added built-in support for snapshot testing which has caused your test to fail. We've never recommended nor suggested that using `jest-snapshot` would be supported which is why we published this as a minor update and not a breaking one.

Glad that it now works for you and I hope your Axe integration is more concise now since you don't need jest snapshot anymore for it. Closing.
